### PR TITLE
recognise STRINGs w/ enclosed interpolations

### PR DESCRIPTION
**Pull request by ZacLN**
This should skip over interpolated regions, though I'm not quite sure if this will behave identically to `julia-parser.scm` when there are syntax errors inside the enclosed code.

### CODE REVIEW

File: src/lexer.jl [hunk 1]
```diff
@@ -637,20 +637,20 @@ end
 
 # Parse a token starting with a quote.
 # A '"' has been consumed
-function lex_quote(l::Lexer)
```
Comment (KristofferC) [inline src/lexer.jl:640]: This will make the function type unstable.

Comment (ZacLN) [reply src/lexer.jl:640]: Good point, I'll have it return an empty token. Related to this, is it worthwhile defining a `const emptytoken = Token()` for cases such as this (I have other instances in my parser) ?

Comment (KristofferC) [reply src/lexer.jl:640]: Yes I think that is worthwhile since we don't have to create the token object then, perhaps (`EMPTY_TOKEN`).

**Comment by ZacLN** [comment]
I've got a fix to push before this is merged

**Pull request merged**